packages/sitecore-jss/src/native-fetcher.ts

```diff
@@ -10,6 +10,10 @@ type NativeDataFetcherOptions = {
    * Override fetch method. Uses native (or polyfilled) fetch by default.
    */
   fetch?: typeof fetch;
+  /**
+   * Optional request timeout.
+   */
+  timeout?: number;
```

Comment:
We have the config, but I don't see the implementation of the timeout for `NativeDataFetcher`? 

We should add tests for our implementation of the timeout for this (and `GraphQLRequestClient`).